src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -598,12 +713,12 @@ internal void ClearCache()
                         User = null
                     };
 
-                    OnBeforeAccess(args);
-                    OnBeforeWrite(args);
+                    OnBeforeAccess(args, null, CacheEvent.TokenTypes.RT);
```

Comment:
Good question. Actually the underlying implementation clears both ATs and RTs, but still, the pre-existing implementation of this Clear() method only triggers OnBeforeAccess callback for one time, rather than 2 times. Which makes sense because the callback is supposed to be triggered only once per API call (to deserialize the token storage). So, it does NOT differenciate AT or RT.

Currently I mapped it to RT because I believe RTs are slightly more important than ATs in this scenarios. So if we can pick only one, we pick RT. Unless, you want to complete remove the token_type field for this scenario?